src/vs/workbench/contrib/terminal/common/terminalService.ts

```diff
@@ -422,10 +422,10 @@ export abstract class TerminalService implements ITerminalService {
 	}
 
 	protected _validateShellPaths(label: string, potentialPaths: string[]): Promise<[string, string] | null> {
-		if (potentialPaths.length === 0) {
+		let current: string;
+		if (potentialPaths.length === 0 || (current = potentialPaths.shift()!) === '') {
```

Comment:
By the change you made, the expected output for ```test: ['', `${system32Path}\\cmd.exe`],``` will be nothing right? 
If one of the `potentialPaths` is empty, it should skip it and continue validation for the remaining paths in the array rather than aborting.

Reply:
@NizamLZ, Nothing all. I think your review is right.